pkg/apis/cloudcredential/v1/types_azure.go

```diff
@@ -40,6 +40,14 @@ type AzureProviderSpec struct {
 	// +optional
 	Permissions []string `json:"permissions,omitempty"`
 
+	AzureClientID string `json:"azureClientID"`
+
+	AzureRegion string `json:"azureRegion"`
+
+	AzureSubscriptionID string `json:"azureSubscriptionID"`
+
+	AzureTenantID string `json:"azureTenantID"`
+
```

Comment:
These need the following:
- comments explaining their usage
- `+optional`
- `omitempty`